### DISCUSSION TITLE
Sensor node and Search node fail installation (1 failure) with Airgap Install. Help!

### DISCUSSION THREAD

**Discussion by AnyrunMalware** [Security-Onion-Solutions/securityonion#5876] [Unsupported Versions] [answered]
I must have installed SO now near 50 times in between all the grids etc... ha ha.

Security Onion 2.3.80 Grid Installation Fails
Node: Keep in mind, this is not a production environment, yet

First node, The management
	- 4xCPU, 16 GB RAM, 200GB Ram
	- Virtual Environment (VMWare)
	- Direct internet access (Don't see any blocks on the FW)
	- Installation completed without errors (SUCCESS!)
	- So-status shows all services running
	- Web console shows node in Grid as "OK, Green"
	- Separate distinct username (local account) for each node local account vs Web Console "email address"

Second node, The search node
	- 4xCPU, 16 GB RAM, 200GB Ram
	- Virtual Environment (VMWare)
	- Web console shows node in Grid as "OK, Green"
	- Separate distinct username (local account) for each node local account 
	- So-status shows all services running
	- If Air Gap: Setup finishes with 1 failure, advises to refer to sosetup.log
	- If Direct Internet Access:  Setup finished without errors, @ hit + [ENTER] for reboot
	- Other than Air Gap vs Direct Internet access, all items are identical for the Search Node.

I use vmware so have the ability to take snapshots and revert back to snapshot to do further testing.

I have access to the sosetup logs for the Manager and Search Node (Both success and failure) (See Attached)

...
[mgmt-sosetup.log](https://github.com/Security-Onion-Solutions/securityonion/files/7340979/mgmt-sosetup.log)
[sensor-success.sosetup.log](https://github.com/Security-Onion-Solutions/securityonion/files/7340980/sensor-success.sosetup.log)
[sensor-failure.sosetup.log](https://github.com/Security-Onion-Solutions/securityonion/files/7340981/sensor-failure.sosetup.log)

**Comment by dougburks** [comment, marked as answer]
sensor-failure.sosetup.log shows `[ERROR   ] 'pkg.clean_metadata': False`. Please run `salt-call pkg.clean_metadata -l debug` after the install fails to see if there's more error information available.

**Reply by AnyrunMalware** [reply]
Hi Doug,
I will run that command (first will need to redeploy and reproduce the error as I have deployed the environment successfully but differently).

It could be due to my lack of understanding on how to setup a Grid with airgapped (sensors and search nodes).

What I was doing, and this was failing for both node types 100pc of the time:
1. Management node w/ direct internet access
2. Search node or Sensor node with airgap access (no internet)
3. = Failure every time.

What worked:
1. Management node with airgap access (no internet)
2. Search node with airgap access (no internet)
3. Sensor node with airgap access (no internet)
4.  = Success on the firs try. (no errors during setup.

I was wondering if you are aware of additional (preferably more detailed) documentation with regards to security onion 2  that describes the grid deployment and how an airgap deployment is managed.

I did find 3 short paragraphs talking about an Airgap deployment but respectfully, I don't think it has all the details. I am left with the impression that you cannot combine airgapped sensors with a management node that has direct internet access but that would not be ideal nor does it make sense?

Some of the questions I am struggling to understand are:
If airgap mode utilized as a grid deployment, how do my sensors obtain the latest suricata rules (for example, for the recent apache 2.2.49 directory traversal vulnerability? Do they get the rules from the management node? 

I will seek to find more documentation describing all of this. 

I appreciate your help with this.

**Reply by dougburks** [reply]
The airgap option is designed for folks that are on totally airgapped networks. This means that NONE of their nodes have any Internet access whatsoever, not even the manager. Folks in this scenario would choose the airgap option for each and every node in their deployment. Updating an airgap deployment is done by sneakernetting the new ISO image across the airgap to the manager and running soup which then updates the entire airgap deployment from that new ISO as described here:
https://docs.securityonion.net/en/2.3/airgap.html

If your manager node is going to have Internet access and you're just trying to prevent other nodes from reaching out to the Internet, then you probably don't need the airgap option at all. You should be able to install your management node in normal Internet mode and then when adding non-manager nodes, select the option for them to pull updates from your manager. This allows those nodes to function properly without having Internet access.

**Reply by AnyrunMalware** [reply]
Doug, thanks for taking the time to answer these important questions.

I am glad to hear reading your last paragraph that there is such an option. (Manager node only has internet, all other nodes go through the manager node for updates).

Unfortunately, the install / setup process only gives these options:
-------------------------------------------------------------------------
How would you like to connect to the Internet ?
1. "Direct" - Internet requests connect directly to the internet.
2. "Direct + Manager" - All traffic passes to the internet normally, but yum updates will instead be pulled from manager.
3. "Proxy" - proxy the traffic for git, docker client, wget, curl, yum, and various other SO components through a separate server in your environment.
4. "Proxy + Manager" - proxy all traffic from the "Proxy" option except yum updates, which will instead pull from the manager.

And reading these options, it gives me the impression that the other nodes still need internet access, even if the yum updates are going through the manager.

I was expecting the option:
------------------------------
 5. All updates are obtained from the manager, no internet access required.
 
 Looking at the documentation, we see the following :
 ----------------------------------------------------------
 Internet Communication
 .
When configuring network firewalls for Internet-connected deployments (non-airgap), **you’ll want to ensure that nodes can connect outbound to the following:**
.
repo.securityonion.net (CentOS Updates)
raw.githubusercontent.com (Security Onion public key)
sigs.securityonion.net (Signature files for Security Onion containers)
ghcr.io (Container downloads)
rules.emergingthreatspro.com (Emerging Threats IDS rules)
www.snort.org (Paid Snort Talos ruleset)
github.com (Strelka and Sigma rules updates)
notary.kolide.co (osquery agent update)
Ubuntu PPAs (OS Updates - Ubuntu only)
download.docker.com (Docker packages - Ubuntu only)
repo.saltstack.com (Salt packages - Ubuntu only)
packages.wazuh.com (Wazuh packages - Ubuntu only)
 .
 .
 .
 Thoughts?

**Reply by dougburks** [reply]
I think the "Direct + Manager" option will work for your use case.

https://docs.securityonion.net/en/2.3/firewall.html#internet-communication is really referring to the deployment as a whole. So if your manager has Internet access and you specify "Direct + Manager" for your other nodes then they should be able to get everything they need from the manager.
